active/0003-route-project-structure.md

```diff
@@ -0,0 +1,152 @@
+│       │   ├── member-permissions.js
+│       │   └── member-permissions.scss
+│       └── admin-route.js
+├── helpers
+│   └── format-date.js
+├── initializers
+│   └── tracking.js
+├── services
+│   └── tracking.js
+├── utils
+│   └── fib.js
+├── shared
+│   ├── dropdown-menu
+│   │   ├── dropdown-menu.js
+|   |   ├── dropdown-menu.scss
+│   └── └── dropdown-menu.hbs
+├── app.js
+└── router.js
+
+```
+
+The largest change here is that `routes` is where majority of your application lives.  In Ember applications, everything is organized and flows from the routes. Because of this it's fitting that routes map back to the project structure.  When your project is organized around the routes several things fall out of this.
+
+- A happy path to coarse bundling of the application
+   - This can be thought as an intermediate step towards engines.
+- Automatic namespacing of components
+- Discrete workspaces for larger teams
+
+## Resolution
+
+The resolution of components and nested-routes is a tiered/fallback approach much like the current pattern in JJ Abrams Resolver.  Routes take precedence over over components.
+
+
+## Coarse Bundling
+
+With a route centric structure, projects can very easily be bundled into coarse bundles. For instance the sample structure described above can very easily be built into the following files.
+
+```
+dist
+├── routes
+│   ├── profile
+│   │   ├── profile.js
+│   │   └── profile.css
+│   └── admin
+│       ├── admin.css
+│       └── admin.js
+└── app.js
+
+```
+
+In this case `app.js` is everything that is top level sans the `routes` directory.  The reason behind this is that these constructs are application agnostic and no nothing about the overall structure of the application.
+
+In an engines world the directories in the `routes` folder can be looked as mountable entry points to the hosting application.
+
+Since you can have nested components sitting the same level, routes once again take precedence over components.
+
+
+## Automatic Namespacing
+
+As of Ember 1.10.0, components can be nested we have the ability to namespace modules based on the route in which the component is contained in. For example, using the `member-photo` component would look like the following.
+
+```
+<profile@member-photo src={{member.imgUrl}} position={{member.position}}"></profile@member-photo>
+``` 
+
+For nested routes you would simply extend the namespace to incorporate the nested route.
+
+```
+<profile.edit@edit-name name={{member.name}}></profile.edit@edit-name>
+```
+
+The convention for components that need to span across more than 1 route is to place them within the `shared` directory.  The shared directory is bundled into `app.js` and is not namespaced as it can appear anywhere in the application.
+
+```
+<dropdown-menu></dropdown-menu>
+```
+
+## Custom Resolver
+
+To enable this type of structure we will need to create a custom resolver and introduce the concept of `@` symbol based namespaces.
```

Comment:
the resolver does currently support namespaces, but requires verbose @ references in the namespaced entities. Rather a curried container that curries the namespace could be used.  I have dinner with @dgeb tonight, maybe we can flesh that out more.

Ultimately, I don't believe namespaces should play a role in this RFC. Rather pods should be merely refined, and isolation concerns should be deferred to engines, or the above private/local helper yielding.